src/md.cc

```diff
@@ -429,10 +454,37 @@ void MGmol<OrbitalsType>::md(OrbitalsType** orbitals, Ions& ions)
         int retval              = 0;
         bool small_move         = true;
         bool last_move_is_small = true;
-        do
+
+#ifdef MGMOL_HAS_LIBROM
+        // variables in ROM MVP solver for Pinned H2O
+        std::vector<double> positions;
+        std::vector<short> anumbers;
+        PinnedH2O H2O_molecule;
+        if (ct.getROMOptions().rom_stage == ROMStage::ONLINE_PINNED_H2O_3DOF)
+        {
+            if (onpe0) os_ << "Rotate Pinned H2O molecule in timestep " << mdstep << std::endl;
+            getAtomicPositions(positions);
+            getAtomicNumbers(anumbers);
+            H2O_molecule.rotate(positions, anumbers);
+            ions.setPositions(positions, anumbers);
+            Potentials& pot = hamiltonian_->potential();
+            pot.initialize(ions);
+            g_kbpsi_->setup(ions);
```

Comment:
@jeanlucf22 Since the `testPinnedH2O_3DOF` complains an identical error about `KBPsiMatrixSparse::computeAll` in debug mode, I think we need to do similar changes in `evaluateDMandEnergyandForces`. My question would be, should we replace `moveVnuc` to `pot.initialize` there as well?